cloud/scope/machinepool.go

```diff
@@ -0,0 +1,518 @@
+
+		// FUTURE: // path.Join("zones", m.Zone(), "diskTypes", string(*disk.DeviceType))
+
+		additionalDisk := &compute.AttachedDisk{
+			AutoDelete: true,
+			InitializeParams: &compute.AttachedDiskInitializeParams{
+				DiskSizeGb:          ptr.Deref(disk.Size, 30),
+				DiskType:            diskType,
+				ResourceManagerTags: shared.ResourceTagConvert(context.TODO(), spec.ResourceManagerTags),
+			},
+		}
+		if strings.HasSuffix(additionalDisk.InitializeParams.DiskType, string(infrav1.LocalSsdDiskType)) {
+			additionalDisk.Type = "SCRATCH" // Default is PERSISTENT.
+			// Override the Disk size
+			additionalDisk.InitializeParams.DiskSizeGb = 375
+			// For local SSDs set interface to NVME (instead of default SCSI) which is faster.
+			// Most OS images would work with both NVME and SCSI disks but some may work
+			// considerably faster with NVME.
+			// https://cloud.google.com/compute/docs/disks/local-ssd#choose_an_interface
+			additionalDisk.Interface = "NVME"
+		}
+		if disk.EncryptionKey != nil {
+			if spec.RootDiskEncryptionKey.KeyType == infrav1.CustomerManagedKey && spec.RootDiskEncryptionKey.ManagedKey != nil {
+				additionalDisk.DiskEncryptionKey = &compute.CustomerEncryptionKey{
+					KmsKeyName: spec.RootDiskEncryptionKey.ManagedKey.KMSKeyName,
+				}
+				if spec.RootDiskEncryptionKey.KMSKeyServiceAccount != nil {
+					additionalDisk.DiskEncryptionKey.KmsKeyServiceAccount = *spec.RootDiskEncryptionKey.KMSKeyServiceAccount
+				}
+			} else if spec.RootDiskEncryptionKey.KeyType == infrav1.CustomerSuppliedKey && spec.RootDiskEncryptionKey.SuppliedKey != nil {
+				additionalDisk.DiskEncryptionKey = &compute.CustomerEncryptionKey{
+					RawKey:          string(spec.RootDiskEncryptionKey.SuppliedKey.RawKey),
+					RsaEncryptedKey: string(spec.RootDiskEncryptionKey.SuppliedKey.RSAEncryptedKey),
+				}
+				if spec.RootDiskEncryptionKey.KMSKeyServiceAccount != nil {
+					additionalDisk.DiskEncryptionKey.KmsKeyServiceAccount = *spec.RootDiskEncryptionKey.KMSKeyServiceAccount
+				}
+			}
+		}
+
+		additionalDisks = append(additionalDisks, additionalDisk)
+	}
+
+	return additionalDisks
+}
+
+// ResourceManagerTags merges ResourceManagerTags from the scope's GCPCluster and GCPMachine. If the same key is present in both,
+// the value from GCPMachine takes precedence. The returned ResourceManagerTags will never be nil.
+func (m *MachinePoolScope) ResourceManagerTags() infrav1.ResourceManagerTags {
+	tags := infrav1.ResourceManagerTags{}
+
+	// Start with the cluster-wide tags...
+	tags.Merge(m.ClusterGetter.ResourceManagerTags())
+	// ... and merge in the Machine's
+	tags.Merge(m.GCPMachinePool.Spec.ResourceManagerTags)
+
+	return tags
+}
+
+// Role returns the machine role from the labels.
+func (m *MachinePoolScope) Role() string {
+	// FUTURE: Or template labels?
+	_, isControlPlane := m.MachinePool.Labels[clusterv1.MachineControlPlaneLabel]
+
+	if isControlPlane {
+		// FUTURE: Extract constants
```

Comment:
TODO for this PR or should we have a follow-up or tracking issue? :-) (also for other `// Future` comments)